### PR TITLE
[CPU][ONNX] Convert FP64 precision to FP32

### PR DESCRIPTION
**Pull request by jiwaszki**
### Details:
 - Add FP64->FP32 conversion in plugin
 - Update ONNX backend xfails

### Tickets:
 - 47330
 - 48100

### CODE REVIEW

File: ngraph/python/tests/test_onnx/test_backend.py [hunk 1]
```diff
@@ -211,10 +210,8 @@ def expect_fail(test_case_path, xfail):  # type: (str) -> None
     (xfail_issue_47323,
      "OnnxBackendPyTorchOperatorModelTest.test_operator_add_broadcast_cpu",
      "OnnxBackendPyTorchOperatorModelTest.test_operator_addconstant_cpu",
-     "OnnxBackendPyTorchOperatorModelTest.test_operator_add_size1_right_broadcast_cpu",),
-    (xfail_issue_47330,
-     "OnnxBackendNodeModelTest.test_mod_mixed_sign_float64_cpu",
-     "OnnxBackendNodeModelTest.test_min_float64_cpu"),
+     "OnnxBackendPyTorchOperatorModelTest.test_operator_add_size1_right_broadcast_cpu",
+     "OnnxBackendNodeModelTest.test_mod_mixed_sign_float64_cpu"),
```
Comment (tomdol) [inline ngraph/python/tests/test_onnx/test_backend.py:214]: This one still fails even with the CPU plugin changes?

**Review by tomdol — APPROVED** [review]

File: ngraph/python/tests/test_onnx/test_backend.py (hunk 1 above)
Comment (jiwaszki) [reply ngraph/python/tests/test_onnx/test_backend.py:214]: That's right, with other error message, so I moved it to other xfail.

**Review by ilyachur — APPROVED** [review]
@dmitry-gorokhov Please have a look

**Comment by dmitry-gorokhov** [comment]
@jiwaszki I just don't understand why do we need this change at all. CPU plug-in doesn't support FP64 precision in general and updating Eltwise node doesn't make any sense. Previous error message seems meaningful enough.
My general point is that CPU nodes shouldn't implicitly handle precisions that a generally not supported by the plug-in.

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp [hunk 2]
```diff
@@ -1087,6 +1087,8 @@ void MKLDNNEltwiseNode::initSupportedPrimitiveDescriptors() {
         } else if (std::find(supportedPrecisions.begin(), supportedPrecisions.end(), prc) == supportedPrecisions.end()) {
             if (prc == Precision::U32 || prc == Precision::I64 || prc == Precision::U64) {
                 return Precision(Precision::I32);
```
Comment (jiwaszki) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp:1089]: > @jiwaszki I just don't understand why do we need this change at all. CPU plug-in doesn't support FP64 precision in general and updating Eltwise node doesn't make any sense. Previous error message seems meaningful enough.
> My general point is that CPU nodes shouldn't implicitly handle precisions that a generally not supported by the plug-in.

@dmitry-gorokhov I see your point, so where should we make these changes? So what about these lines? I see converting of I64 to I32, isn't that a same thing? I think this approach of fixing precision issues was used for a while.

File: ngraph/python/tests/test_onnx/test_backend.py (hunk 1 above)
Comment (jiwaszki) [reply ngraph/python/tests/test_onnx/test_backend.py]: Now both of them are passing.

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp (hunk 2 above)
Comment (jiwaszki) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp:1089]: Discussed offline.

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp [hunk 3]
```diff
@@ -1087,6 +1087,8 @@ void MKLDNNEltwiseNode::initSupportedPrimitiveDescriptors() {
...
+            } else if (prc == Precision::FP64) {
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp:1090]: Looks like we don't need this change anymore?

**Review by GlebKazantaev — APPROVED** [review]

**Review by postrational — APPROVED** [review]

**Comment by postrational** [comment]
@dmitry-gorokhov 
> I just don't understand why do we need this change at all. CPU plug-in doesn't support FP64 precision in general and updating Eltwise node doesn't make any sense. Previous error message seems meaningful enough.
> My general point is that CPU nodes shouldn't implicitly handle precisions that a generally not supported by the plug-in.

I guess the answer to your question is that this change allows the following ONNX compliance tests to pass:

* "OnnxBackendNodeModelTest.test_mod_mixed_sign_float64_cpu"
* "OnnxBackendNodeModelTest.test_min_float64_cpu"
* "OnnxBackendNodeModelTest.test_eyelike_with_dtype_cpu"

If you have a suggestion how to make these things pass in a different way, please let us know.

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**